File: server/order-service/src/main/java/com/sudo248/orderservice/controller/order/dto/UserDto.java
package com.sudo248.orderservice.controller.order.dto;

import com.sudo248.orderservice.repository.entity.order.Gender;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.util.Date;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class UserDto {
    private String userId;

    private String fullName;

    private String emailOrPhoneNumber;

    private LocalDate dob;

    private String bio;

    private String avatar;

    private String cover;

    private AddressDto address;

    private Gender gender;
}

File: client/mobile/Sudoo/app/src/main/java/com/sudo248/sudoo/data/dto/user/UserDto.kt
package com.sudo248.sudoo.data.dto.user

import com.sudo248.sudoo.domain.entity.auth.Role
import com.sudo248.sudoo.domain.entity.user.Gender
import java.util.Date

data class UserDto(
    val userId: String,
    val fullName: String,
    val emailOrPhoneNumber: String,
    val dob: Date,
    val bio: String = "",
    val avatar: String,
    val cover: String = "",
    val address: AddressDto,
    val gender: Gender,
    val role: Role = Role.CONSUMER
)
